### PR TITLE
review realtime socket

### PR DESCRIPTION
**Pull request by hiendt-0025**
Ticket : https://edu-redmine.sun-asterisk.vn/issues/29030
WHAT : Review realtime socket
HOW:
review

### CODE REVIEW

**Review by huutuan2308 — APPROVED** [review]

**Review by CoBeKieuKy — APPROVED** [review]

**Comment by hiendt-0025** [comment]
ready

File: foodanddrink/restaurant/consumers.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        content = text_data_json['content']
+        vote = float(text_data_json['vote'])
+        productId = text_data_json['productId']
+        userId = text_data_json['userId']
+
+        user = Customer.objects.get(user_id=userId)
+        product = Product.objects.get(id=productId)
+
+        review = Review(user = user, product = product, content = content, vote = vote )
+        review.save()
```
Comment (thaodtd-0665) [inline foodanddrink/restaurant/consumers.py]: check valid data lúc save nha em

File: foodanddrink/restaurant/consumers.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        rate = Review.objects.filter(product=product).aggregate(Avg('vote'))
+        product.vote = list(rate.values())[0]
+        product.save()
```
Comment (thaodtd-0665) [inline foodanddrink/restaurant/consumers.py]: check chỗ này nữa

**Comment by hiendt-0025** [comment]
ready

**Review by thaodtd-0665 — APPROVED** [review]

**Pull request merged**